dmoj/judge.py

```diff
@@ -284,22 +226,235 @@ def listen(self):
         self.updater.start()
         self.packet_manager.run()
 
-    def __enter__(self):
-        return self
-
-    def __exit__(self, exception_type, exception_value, traceback):
-        pass
-
     def murder(self):
         """
         End any submission currently executing, and exit the judge.
         """
-        self.terminate_grading()
+        self.abort_grading()
         self.updater_exit = True
         self.updater_signal.set()
         if self.packet_manager:
             self.packet_manager.close()
 
+    def log_internal_error(self, exc=None, message=None):
+        if not message:
+            # If exc is exists, raise it so that sys.exc_info() is populated with its data
+            if exc:
+                try:
+                    raise exc
+                except KeyboardInterrupt:
+                    # Let KeyboardInterrupt bubble up.
+                    raise
+                except:  # noqa: E722, we want to catch everything
+                    pass
+
+            exc = sys.exc_info()
+            message = ''.join(traceback.format_exception(*exc))
+
+        # Strip ANSI from the message, since this might be a checker's CompileError
+        # ...we don't want to see the raw ANSI codes from GCC/Clang on the site.
+        # We could use format_ansi and send HTML to the site, but the site doesn't presently support HTML
+        # internal error formatting.
+        self.packet_manager.internal_error_packet(strip_ansi(message))
+
+        # Logs can contain ANSI, and it'll display fine
+        print(message, file=sys.stderr)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        pass
```

Comment:
Is there even a point to this context manager if it doesn't do any cleanup? i.e. now, `with Judge() as judge:` is the same as `judge = Judge()`.

Reply:
Yeah, removed.